atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/BackgroundSweeperFastTest.java

```diff
@@ -92,6 +93,117 @@ public void testWritePriorityAfterSecondRunCompletesSweep() {
+                eq(ImmutableSweepProgress.builder()
+                        .tableRef(TABLE_REF)
+                        .staleValuesDeleted(5)
+                        .cellTsPairsExamined(21)
+                        .minimumSweptTimestamp(4567L)
+                        .startRow(new byte[] {4, 5, 6})
+                        .startColumn(PtBytes.toBytes("unused"))
+                        .timeInMillis(30L)
+                        .startTimeInMillis(50L)
+                        .build()));
+    }
+
+    @Test
+    public void testSecondRunMaySweepDifferentTable() {
+        setNoProgress();
+        setNextTableToSweep(TABLE_REF);
+
+        setupTaskRunner(ImmutableSweepResults.builder()
+                .staleValuesDeleted(2)
+                .cellTsPairsExamined(10)
+                .minSweptTimestamp(12345L)
+                .nextStartRow(Optional.of(new byte[] {1, 2, 3}))
+                .timeInMillis(10L)
+                .timeSweepStarted(50L)
+                .build());
+        backgroundSweeper.runOnce();
+
+        ImmutableSweepProgress progressAfterFirstIteration = ImmutableSweepProgress.builder()
+                .tableRef(TABLE_REF)
+                .staleValuesDeleted(2)
+                .cellTsPairsExamined(10)
+                .minimumSweptTimestamp(12345L)
+                .startRow(new byte[] {1, 2, 3})
+                .startColumn(PtBytes.toBytes("unused"))
+                .timeInMillis(10L)
+                .startTimeInMillis(50L)
+                .build();
+        Mockito.verify(progressStore).saveProgress(
+                eq(progressAfterFirstIteration));
+        setProgress(progressAfterFirstIteration);
+
+        // Between iterations 1 and 2, OTHER_TABLE gets added to the priority list.
+        setNoProgress(OTHER_TABLE);
+        setupTaskRunner(OTHER_TABLE, ImmutableSweepResults.builder()
+                .staleValuesDeleted(3)
+                .cellTsPairsExamined(11)
+                .minSweptTimestamp(4567L)
+                .nextStartRow(Optional.of(new byte[] {4, 5, 6}))
+                .timeInMillis(20L)
+                .timeSweepStarted(50L)
+                .build());
+        overrideConfig = ImmutableSweepPriorityOverrideConfig.builder()
+                .addPriorityTables(OTHER_TABLE.getQualifiedName())
+                .build();
+        setNextTableToSweep(OTHER_TABLE);
+        backgroundSweeper.runOnce();
+
+        ImmutableSweepProgress progressAfterSecondIteration = ImmutableSweepProgress.builder()
+                .tableRef(OTHER_TABLE)
+                .staleValuesDeleted(3)
+                .cellTsPairsExamined(11)
+                .minimumSweptTimestamp(4567L)
+                .startRow(new byte[] {4, 5, 6})
+                .startColumn(PtBytes.toBytes("unused"))
+                .timeInMillis(20L)
+                .startTimeInMillis(50L)
+                .build();
+        Mockito.verify(progressStore).saveProgress(
+                eq(progressAfterSecondIteration));
+    }
+
```

Comment:
Tests look good. Is there a way we can verify the states of timelock / the lock service at that time?